docs/rfcs/XXXX_storage_backends.md

```diff
@@ -0,0 +1,288 @@
+
+### Study Buckets
+
+Studies are central entities in the the application. They partition data that is collected and submitted by different research projects and investigators. For this reason, it was determined that each study would have its own bucket to segment it from other collections of data.
+
+#### Creation
+
+The the application automatically creates and configures buckets upon creation of a new study. These buckets are created in the same account as the IAM role that the API uses (Kids First Open Data). This account happens to be different from the account where the API actually resides (Kids First STRIDES). This is necessary as compute used to be contained within Kids First Open Data but was migrated to the STRIDES account. However, the data did not move and the codebase required that the IAM used was owned by the same account as the data.
+
+#### Configuration
+
+Each study bucket needs to be configured with the correct security policies, encryption, replication, logging, and so on. For this reason, the bucket creation process was automated and included as part of the the application's service offerings. Upon creation, the the application will configure these additional settings ensuring all buckets are setup correctly.
+
+#### Layout
+Each study bucket has a similar layout:
+```
+source/
+  uploads/
+harmonized/
+```
+
+Under `source` are data files such as raw read files received from the sequencing center, MRI DICOMs from the investigator, and so forth. The `uploads` directory contains files that are uploaded from the the application. `harmonized` contains files that were processed by the BIX team using things such as a standard alignment workflow or calling pipeline.
+
+### Issues with Study Buckets
+
+Study buckets are restricted to the same account as the IAM profile assigned to the application's compute instance.
+The IAM profile contains a large surface of permissions needed to create and modify buckets.
+
+## Storage Backend Solution
+By allowing users to create "Storage Backends" that grant the application access to bucket resources in other accounts, we will be able to provide more flexibility in how organizations choose to store their data.
+
+## Default Storage
+![enter image description here](https://i.ibb.co/vBb1YsN/storage-layout.png)
+"Default Storage" will refer to the defaults for how a new organization and its studies will be stored. It's important that new organizations and studies be quickly bootstraped into a working state without unnecessary configuration. To achieve this, the application will provide a default workspace for documents to land. This will be achieved by providing a Storage Backend that will point to the application's application bucket.
```

Comment:
> It's important that new organizations and studies be quickly bootstraped into a working state without unnecessary configuration

just curious but why is it important for studies to be bootstrapped quickly? won't that lead to files being uploaded into the wrong bucket once the study is properly setup? and then they will need to migrate the data from the default bucket into their bucket - or re-upload everything

Reply:
Yes, the files would flow to potentially the wrong location if the study was setup without the storage backend ready to go.
The ability to quickly setup and start uploading to a study is more about stream-lining the process for the administrative user. In our case, the user creating the study is not the one with the know-how to setup and configure the storage backend. This leads to a blocker in the time-to-first-upload process.

However, another benefit of the bootstrapping was also that we could ensure each study bucket was being configured identically. Now that we are moving to a user-managed model for storage backends, this is less of a benefit.

Because using the 'default storage backend' will be hostile to users who will need to move to a new backend, perhaps we should not allow the use of a default until we also allow seamless migration. Maybe we start with a more restrictive feature set and aim to add this 'default' later as the next iteration:
1) Require users to select or create a new storage backend when creating their study
2) Allow users to select a 'default storage backend' when creating their study and enable seamless migration of data when the user changes their study's backend after it's been created.